allenact/base_abstractions/distributions.py

```diff
@@ -44,6 +44,24 @@ class CategoricalDistr(torch.distributions.Categorical, Distr):
     def mode(self):
         return self._param.argmax(dim=-1, keepdim=False)  # match sample()'s shape
 
+    def log_prob(self, value: torch.Tensor):
+        if value.shape == self.logits.shape[:-1]:
+            return super(CategoricalDistr, self).log_prob(value=value)
+        elif value.shape == self.logits.shape[:-1] + (1,):
+            return (
+                super(CategoricalDistr, self)
+                .log_prob(value=value.squeeze(-1))
+                .unsqueeze(-1)
+            )
+        else:
+            raise NotImplementedError(
```

Comment:
I guess this has been added to take care of the possibility to introduce a bug in the on-policy losses we discussed a few days ago. I still don't see the possible error, but I trust this is the right thing to do 👍